nnf/__init__.py

```diff
@@ -998,6 +998,53 @@ def pair(node: NNF) -> NNF:
 
         return pair(sentence)
 
+    def project(self: T_NNF, names: 't.FrozenSet[Name]') -> 'NNF':
+        """Dual of :meth:``forget``: will forget all variables not given"""
+        return self.forget(self.vars() - names)
+
+    def forget_aux(self: T_NNF) -> 'NNF':
```

Comment:
The `T_NNF` annotation is superfluous here and everywhere else you added it. It's only useful if you use it multiple times in the signature.

Reply:
I'd assumed something was required modelling after the following method (`deduplicate`). Seems to be fine with them removed, yep.

Reply:
`.deduplicate()` always returns exactly the same type you put in, but we don't have a relation like that here.

Even there it's not required, it's just so the typechecker could pick up that if you deduplicate (say) a CNF sentence it's still CNF afterwards. The method itself would still pass mypy without it.